mirrord-agent/src/sniffer.rs

```diff
@@ -236,12 +235,17 @@ pub async fn packet_worker(
     mut rx: Receiver<SnifferCommand>,
     interface: String,
     container_id: Option<String>,
+    container_runtime: Option<String>,
 ) -> Result<()> {
     debug!("setting namespace");
-    if let Some(container_id) = container_id {
-        let namespace = get_container_namespace(container_id).await?;
-        debug!("Found namespace to attach to {:?}", &namespace);
-        set_namespace(&namespace)?;
+    if let (Some(container_id), Some(container_runtime)) = (container_id, container_runtime) {
+        let runtime: Box<dyn Namespace> = match container_runtime.as_str() {
```

Comment:
I think we can have better design here than using Box + dyn. 
I'm trying to create an example, maybe @meowjesty will have an idea.